src/conf_mode/service_stunnel.py

```diff
@@ -0,0 +1,223 @@
+                    and 'connect' in app_conf:
+                raise ConfigError("The 'connect' option cannot be used with the 'socks' protocol in server mode.")
+
+            for item in ['connect', 'listen']:
+                if item in app_conf:
+                    if 'port' not in app_conf[item]:
+                        raise ConfigError(f'{mode} [{app}]: {item} port number is required!')
+                elif item == 'listen':
+                    raise ConfigError(f'{mode} [{app}]: {item} port number is required!')
+
+            if 'options' in app_conf:
+                protocol = app_conf.get('protocol')
+                if protocol not in ['connect', 'smtp']:
+                    raise ConfigError("Additional option is only supported in the 'connect' and 'smtp' protocols.")
+                if protocol == 'smtp' and ('domain' in app_conf['options'] or 'host' in app_conf['options']):
+                    raise ConfigError("Protocol 'smtp' does not support options 'domain' and 'host'.")
+
+                # set default authentication option
+                if 'authentication' not in app_conf['options']:
+                    app_conf['options']['authentication'] = 'basic' if protocol == 'connect' else 'plain'
+
+                for option, option_config in app_conf['options'].items():
+                    if option == 'authentication':
+                        if protocol == 'connect' and option_config not in ['basic', 'ntlm']:
+                            raise ConfigError("Supported authentication types for the 'connect' protocol are 'basic' or 'ntlm'")
+                        elif protocol == 'smtp' and option_config not in ['plain', 'login']:
+                            raise ConfigError("Supported authentication types for the 'smtp' protocol are 'plain' or 'login'")
+                    if option == 'host':
+                        if 'address' not in option_config:
+                            raise ConfigError('Address is required for option host.')
+                        if 'port' not in option_config:
+                            raise ConfigError('Port is required for option host.')
+
+            # check pki certs
+            for key in ['ca_certificate', 'certificate']:
+                tmp = dict_search(f'ssl.{key}', app_conf)
+                if mode == 'server' and 'ca_' not in key and not tmp and 'psk' not in app_conf:
+                    raise ConfigError(f'{mode} [{app}]: TLS server needs a certificate or PSK')
+                if tmp:
+                    if 'ca_' in key: verify_pki_ca_certificate(stunnel, tmp)
+                    else: verify_pki_certificate(stunnel, tmp)
+
+            #check psk
+            if 'psk' in app_conf:
+                for psk, psk_conf in app_conf['psk'].items():
+                    if 'id' not in psk_conf or 'secret' not in psk_conf:
+                        raise ConfigError(
+                            f'Authentication psk "{psk}" missing "id" or "secret"')
+
```

Comment:
Verify is port already bonded by another service
```
set service ssh port 22
set service stunnel server my-server listen port 22
```

Reply:
done